pkg/innerring/subnet.go

```diff
@@ -315,5 +319,93 @@ func (s *Server) handleSubnetRemoval(e event.Event) {
 		return
 	}
 
-	// TODO: #1162 handle removal of the subnet in netmap candidates
+	// handle subnet changes in netmap
+
+	candidates, err := s.netmapClient.GetCandidates()
+	if err != nil {
+		s.log.Error("getting netmap candidates",
+			zap.Error(err),
+		)
+
+		return
+	}
+
+	var removedID subnetid.ID
+	_ = removedID.Unmarshal(delEv.ID()) // parsing errors were checked above
```

Comment:
Too risky IMO. We need to get `removeID` from the place where it was checked `above`, or check for errors explicitly.

Reply:
ok, agree